README.md

```diff
@@ -23,33 +23,95 @@ Gluon is a small, statically-typed, functional programming language designed for
+factorial 10
+```
 
-```toml
-[dependencies]
-gluon = "0.6.2"
+### Syntax
+
+Larger example which display most if not all of the syntactical elements in the language.
+
+```f#,rust
+let io = import! std.io
+
+// `let` declares new variables.
+let id x = x
+
+let factorial n =
+        if n < 2
+        then 1
+        else n * factorial (n - 1)
+
+// `type` is used to declare a new type.
+// In this case we declare `Countable` to be a record with a single field (count) which is a function
+// taking a single argument and returning an integer
+type Countable a = { count : a -> Int }
+
+// "Counting" an integer just means returning the integer itself
+let countable_Int : Countable Int = { count = \x -> x }
+
+let list_module =
+    // Declare a new type which only exists in the current scope
+    type List a = | Cons a (List a) | Nil
+    let map f xs =
+            match xs with
+                | Cons y ys -> Cons (f y) (map f ys)
+                | Nil -> Nil
+    // Define a count instance over lists which counts each of the elements and sums
+    // the results
+    let countable_List c : Countable a -> Countable (List a) =
+        let count xs =
+            match xs with
+            | Cons y ys -> c.count y + count ys
+            | Nil -> 0
+        { count }
+    {
+        // Since `List` is local we export it so its constructors can be used
+        // outside the current scope
+        List,
+        countable_List,
+        map
+    }
+
+// Bring the `List` type and its constructors into scope
+let { List, countable_List } = list_module
+
+// Create a `Countable` record for `List Int`
+let { count } : Countable (List Int) = countable_List countable_Int
+
+if count (Cons 20 (Cons 22 Nil)) == 41 then
+    error "This branch is not executed"
+else
+    // `import! <filename>` loads a module stored at `filename`
+    let io = import! std.io
+    io.print "Hello world!"
 ```
 
-#### Other languages
-Currently the easiest way to interact with the gluon virtual machine is through Rust but a rudimentary [C api][] exists which will be extended in the future to bring it closer to the Rust api.
+## Getting started
 
-[C api]: https://github.com/gluon-lang/gluon/blob/master/c-api/src/lib.rs
+### Try online
 
-## Tools
+You can try gluon in your browser at the [try_gluon](http://52.28.135.57:8080/) server. ([Github](https://github.com/gluon-lang/try_gluon))
 
 ### REPL
 
-Gluon has a small executable which can be used to run gluon programs directly or in a small REPL. The REPL can be started by passing the `-i` flag to the built repl executable which can be run with `cargo run -- -i`.
+Gluon has a small executable which can be used to run gluon programs directly or in a small REPL. The REPL can be started by passing the `-i` flag to the built repl executable which can be run with `cargo run -p gluon_repl -- -i`.
```

Comment:
👍 The repl used to be located in the main crate but I forgot to update this to match when I moved it.